### ISSUE TITLE
CI: system tests: make random_free_port() parallel-safe

### ISSUE THREAD

**Issue by edsantiago** [containers/podman#23488]
I thought I had, but [guess not](https://api.cirrus-ci.com/v1/artifact/task/6446991591342080/html/sys-remote-fedora-40-rootless-host-sqlite.log.html#t--00178p). This is a placeholder issue until I get it right.

**Comment by edsantiago** [comment]
It's a Bats bug: parallelization doesn't work the way I expected it to (not a bug) and there's no documentation about it nor any way to get a job slot number (yes a bug). Filed https://github.com/bats-core/bats-core/pull/968

**Comment by edsantiago** [comment]
New mechanism submitted: https://github.com/containers/podman/pull/23275/commits/7c3d294ec6b680b7cb0fff63bf8303741d3eac05

Tested on my laptop, because CI is down. No failures seen yet.

**Comment by Luap99** [comment]
I am somewhat sure this fixes https://github.com/containers/podman/issues/23471 as well.
But I will test locally as well as it is was simple to reproduce

**Comment by Luap99** [comment]
Before this it failed basically every time for me, now it no longer fails after 10+ runs locally

All I see in from current PR is
```
cp: cannot create regular file '/tmp/podman-bats-logs.WUKB1OTFB/7-podman puts pasta IP in %2Fetc/hosts.log': No such file or directory
```
which seems to cause exit code 1 even though bats prints `86 tests, 0 failures, 40 skipped in 15 seconds`

**Comment by edsantiago** [comment]
Are you on f40 and have you dnf-upgraded? This is fixed in bats-11

**Comment by edsantiago** [comment]
Took a little longer than I expected (i.e. more than two runs), but:
```console
$ while :;do ./bats -T --rootless --tag='ci:parallel' 505 || break;done  
...
✗ |505| TCP port range forwarding, IPv4, tap [131778]                                                                                                          
   tags: ci:parallel                                                                                                                                            
   (from function `bail-now' in file test/system/helpers.bash, line 189,                                                                                        
    from function `die' in file test/system/helpers.bash, line 937,                                                                                             
    from function `run_podman' in file test/system/helpers.bash, line 539,                                                                                      
    from function `pasta_test_do' in file test/system/505-networking-pasta.bats, line 235,                                                                      
    in test file test/system/505-networking-pasta.bats, line 483)                                                                                               
     `pasta_test_do' failed                                                                                                                                     
                                                                                                                                                                
   [06:08:58.901317437] $ /home/esm/src/atomic/2018-02.podman/libpod/bin/podman info --format {{.Host.Pasta.Executable}}                                        
   [06:09:00.064937823] /usr/bin/pasta                                                                                                                          
                                                                                                                                                                
   [06:09:00.318915429] $ /home/esm/src/atomic/2018-02.podman/libpod/bin/podman run --rm --name=c-socat-t21-6gost7x6 --net=pasta -p [192.168.101.31]:5730-5732:5
730-5732/tcp quay.io/libpod/testimage:20240123 sh -c for port in $(seq 5730 5732); do                              socat -u TCP4-LISTEN:${port},bind=[192.168.10
1.31] STDOUT &                          done; wait                                                                                                              
   [06:11:10.344876219] timeout: sending signal TERM to command ‘/home/esm/src/atomic/2018-02.podman/libpod/bin/podman’                                         
   timeout: sending signal KILL to command ‘/home/esm/src/atomic/2018-02.podman/libpod/bin/podman’                                                              
   [06:11:10.348823337] [ rc=137 (** EXPECTED 0 **) ]                                                                                                           
   #/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv                                                                                                         
   #| FAIL: exit code is 137; expected 0                                                                                                                        
   #\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                         
   # [teardown]
```

**Comment by Luap99** [comment]
I am on f39 and there is no bats update there. Anyhow I just renamed the test for now and running this in a loop for 15 minutes without failure. Before that it failed on almost every run so I am very certain that the the port conflict caused the hangs in the udp case due the incorrect behavior of REUSEADDR. https://github.com/containers/podman/issues/23471#issuecomment-2265138841

I now also got the TCP hang after 15 minutes but this seems different from the udp hangs listed in https://github.com/containers/podman/issues/23471 I think.
